core/src/lib.rs

```diff
@@ -71,97 +60,79 @@ use temporal_sdk_core_protos::{
     temporal::api::history::v1::History,
 };
 
-#[cfg(test)]
-use crate::test_help::MockWorker;
-
 lazy_static::lazy_static! {
     /// A process-wide unique string, which will be different on every startup
     static ref PROCCESS_UNIQ_ID: String = {
         uuid::Uuid::new_v4().to_simple().to_string()
     };
 }
 
-/// Holds various configuration information required to call [init]
-#[derive(Debug, Clone, derive_builder::Builder)]
-#[builder(setter(into))]
-#[non_exhaustive]
-pub struct CoreInitOptions {
-    /// Options for the connection to the temporal server
-    pub gateway_opts: ServerGatewayOptions,
-    /// Options for telemetry (traces and metrics)
-    #[builder(default)]
-    pub telemetry_opts: TelemetryOptions,
-}
-
-/// Initializes an instance of the core sdk and establishes a connection to the temporal server.
-/// Expects that a tokio runtime exists.
-pub async fn init(opts: CoreInitOptions) -> Result<CoreSDK, CoreInitError> {
-    telemetry_init(&opts.telemetry_opts).map_err(CoreInitError::TelemetryInitError)?;
-    // Initialize server client
-    let server_gateway = opts.gateway_opts.connect(Some(&METRIC_METER)).await?;
-
-    Ok(CoreSDK::new(server_gateway, opts))
+/// Initialize a worker bound to a task queue
+pub fn init_worker<SG: ServerGatewayApis + Send + Sync + 'static>(
+    worker_config: WorkerConfig,
+    client: SG,
+) -> Worker {
+    init_worker_pre_arcd(worker_config, Arc::new(client))
 }
 
-/// Initialize core using a provided gateway instance, which is typically a mock
-pub fn init_mock_gateway<SG: ServerGatewayApis + Send + Sync + 'static>(
-    opts: CoreInitOptions,
-    server_gateway: SG,
-) -> Result<CoreSDK, CoreInitError> {
-    telemetry_init(&opts.telemetry_opts).map_err(CoreInitError::TelemetryInitError)?;
-    Ok(CoreSDK::new(server_gateway, opts))
+/// Initialize a worker bound to a task queue, if the client is already behind an `Arc`.
+// Ideally this would be done with `impl Into<Arc<SG>>` but that imposes type clarifications
+// at the callsite sometimes.
+pub fn init_worker_pre_arcd<SG: ServerGatewayApis + Send + Sync + 'static>(
```

Comment:
I'd be ok if you wanted to always require an Arc and just get rid of the non-Arc overload